### PR TITLE
DBZ-2800 Fix AWS user roles validation

### PR DESCRIPTION
**Pull request by rk3rn3r**
closes https://issues.redhat.com/browse/DBZ-2800

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java [hunk 1]
```diff
@@ -120,11 +120,30 @@ public Config validate(Map<String, String> connectorConfigs) {
...
+                                    " FROM pg_roles r WHERE r.rolname = current_user",
+                            connection.singleResultMapper(rs -> rs.getBoolean("rolcanlogin")
+                                    && (rs.getBoolean("rolreplication")
+                                            || rs.getBoolean("aws_superuser")
+                                            || rs.getBoolean("aws_admin")
+                                            || rs.getBoolean("aws_repladmin")),
+                                    "Could not fetch roles"))) {
                         final String errorMessage = "Postgres roles LOGIN and REPLICATION are not assigned to user: " + connection.username();
                         logger.error(errorMessage);
-                        userResult.addErrorMessage(errorMessage);
```
Comment (rk3rn3r) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:127]: Removing the validation error. But imho AWS is now fully supported. Maybe we can bring back that with the next release? WDYT?

Comment (jpechane) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:127]: In that case I'd vot for introducing an internal (undocumented) config option that could override that behaviour - like in `io.debezium.relational.history.DatabaseHistory.DDL_FILTER`

Comment (rk3rn3r) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:127]: Good point! I created https://issues.redhat.com/browse/DBZ-2893 and https://issues.redhat.com/browse/DBZ-2894 to handle this properly after tomorrow's release.

**Comment by gunnarmorling** [comment]
Failures in PG tests are unrelated?

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java [hunk 2]
```diff
@@ -120,11 +120,30 @@ public Config validate(Map<String, String> connectorConfigs) {
                     }
                     // check user for LOGIN and REPLICATION roles
                     if (!connection.queryAndMap(
-                            "SELECT rolcanlogin, rolreplication FROM pg_roles WHERE rolname = current_user",
-                            connection.singleResultMapper(rs -> rs.getBoolean("rolcanlogin") && rs.getBoolean("rolreplication"), "Could not fetch roles"))) {
+                            "SELECT r.rolcanlogin AS rolcanlogin, r.rolreplication AS rolreplication," +
```
Comment (jpechane) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:123]: How about using query like `SELECT rolcanlogin, rolreplication FROM pg_roles WHERE rolname = current_user UNION ALL SELECT rolcanlogin, rolreplication FROM pg_catalog.pg_auth_members m JOIN pg_catalog.pg_roles b ON (m.roleid = b.oid) WHERE oid=(SELECT oid FROM pg_roles WHERE rolname = current_user)` and check there is at least one `true` for each of the roles?

Comment (rk3rn3r) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:123]: Wouldn't this only work when we are on AWS, but then will fail when we are not? Had a hard time to figure a smaller & nicer, less repetitive query. Also, a proper solution needs to check a role tree (like a role is assigned to another role and another role and this role finally to the user). This should be handled algorithmically and not with a complex query with https://issues.redhat.com/browse/DBZ-2894 .

**Comment by rk3rn3r** [comment]
@gunnarmorling yes the test failure is unrelated. Still some unreliability with GH Actions tests.

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java (hunk 2 above)
Comment (jpechane) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresConnector.java:123]: It should work for both, that's while there is the `UNION`. IMHO it just really does not cover the inheritance

**Pull request merged**